crates/redeem/src/tests.rs

```diff
@@ -526,3 +526,59 @@ fn test_set_redeem_period_only_root() {
         assert_ok!(Redeem::set_redeem_period(Origin::root(), 1));
     })
 }
+
+mod spec_based_tests {
+    use super::*;
+
+    #[test]
+    fn test_request_reduces_to_be_replaced() {
+        // Checked POSTCONDITION: `decreaseToBeReplacedTokens` MUST be called, supplying `vault` and `burnedTokens`.
+        // The returned `replaceCollateral` MUST be released by this function.
+        run_test(|| {
+            let amount_to_redeem = 100;
+            let to_be_replaced_tokens = 10;
+            let replace_collateral = 100;
+            assert_ok!(<Test as vault_registry::Config>::Wrapped::mint(
+                &ALICE,
+                amount_to_redeem
+            ));
+            <vault_registry::Pallet<Test>>::insert_vault(
```

Comment:
Probably this should be mocked as well, otherwise it's kinda weird to instead mock the other functions called on the vault-registry.

Reply:
You mean mock the vault insertion? I'm mocking `decrease_to_be_replaced_tokens` and `release_collateral` because these calls need to be checked as part of the postcondition. The properties of the inserted vault are picked such that the mocked functions will have the wrong parameters if there's a bug. I'm not sure how to mock this.

Reply:
No, mock the return from whatever getter it uses, but nevermind this is not a big deal.